src/main/java/com/example/shoppingonlybackend/repository/ProductRepository.java

```diff
@@ -2,7 +2,15 @@
 
 import com.example.shoppingonlybackend.entity.Product;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
 
 public interface ProductRepository  extends JpaRepository<Product, Integer> {
+       List<Product> findByNameContaining(String keyword);
+
+    List<Product> findByPriceGreaterThan(double price);
```

Comment:
```suggestion
    List<Product> findByPriceGreaterThan(Double price);
```
Mümkün olduğunda metot parametreleri primitive tip olmasın.